### DISCUSSION TITLE
[Edit this page] points to /master/docs/ regardless of mkdocs.yml config

### DISCUSSION THREAD

**Discussion by ludwikc** [mkdocs/mkdocs#3801] [Q&A]
When enabling [Edit this page on Github] feature, the link points to 

`/master/docs/file.md` regardless of mkdocs.yml configuration, generating 404 error. 

both `branch` and `docs_dir` should be read from mkdocs.yml and links generated accodingly. 

![SIADLAK_20240806-Wte1LOxG](https://github.com/user-attachments/assets/579fbdec-d047-4914-9dab-44c80c750571)

For the record, the proper file path (in this particular case) should be: `main/ludwikc-pov/00_StronaGłówna.md`

**Comment by pawamoy** [comment]
Hi @ludwikc, thanks for the report :slightly_smiling_face: 

Is it possible that you were still on branch master when building your docs? :thinking:

**Reply by ludwikc** [reply]
No, this is a brand new repo and brand new build. 

And `ludwikc-pov` was my docs_dir from the very beginning.

**Comment by ludwikc** [comment]
@tomchristie you’ve converted this thread to discussion, although i still believe it’s a bug. what’s to discuss in here?

**Reply by pawamoy** [reply]
@ludwikc are you able to provide a reproducible example of the described issue?

**Reply by kamilkrzyskow** [reply]
Hello @ludwikc,
don't know about the Tom's reasoning, but this change allowed me to see the discussion, as I get (*sometimes*) notified about discussions but not issues :v:

Since you're using the `material` theme you could follow this guide to create a minimal reproduction of your issue, as the `edit link` functionality is handled by templates in the theme:
- https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/
- https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/actions.html

I don't think there is a bug in neither MkDocs or the theme templates, so maybe something else (like a plugin) changes the value of the `page.edit_url`? But first of all there is no `edit_uri` in your `mkdocs.yml` file set to `edit/main/ludwikc-pov/` and there is also another issue with your configuration file, the [`site_url`](https://github.com/ludwikc/ludwikc.github.io/blob/6f0411a57a53b2cde9b8de84621056cef0ce2762/mkdocs.yml#L2) uses a wrong domain name, which makes the canonical links invalid.

Your docs root directory `ludwikc-pov/` lacks an `index.md` file to redirect to your `/home/` path. You can use the meta template setting together with the builtin `redirect.html` from the Material theme:
- https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-template
- https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/redirect.html

`theme.language` is set to `en` instead of `pl`, which influences the tooltips in the Material theme
